### PR TITLE
Libisis now installs based on version string

### PR DESCRIPTION
**Pull request by Kelvinrr**
## Description

libisis still has a 3.x.x suffix, this fixes it to be based on the current ISIS version. 

## Related Issue

https://github.com/USGS-Astrogeology/ISIS3/issues/3365

## How Has This Been Tested?

full local build and conda build have correct libraries

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [AUTHORS.rst](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/AUTHORS.rst) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/cmake/cmake_install.cmake [hunk 1]
```diff
@@ -31,20 +31,3 @@ endif()
...
-  EXECUTE_PROCESS(COMMAND ln -sf libisis3.5.0.so /scratch/isiscmake/isis/build/lib/libisis3.5.so)
-endif()
-
-if("${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified" OR NOT CMAKE_INSTALL_COMPONENT)
-  EXECUTE_PROCESS(COMMAND ln -sf libisis3.5.so /scratch/isiscmake/isis/build/lib/libisis3.so)
-endif()
-
-if("${CMAKE_INSTALL_COMPONENT}" STREQUAL "Unspecified" OR NOT CMAKE_INSTALL_COMPONENT)
-  EXECUTE_PROCESS(COMMAND ln -sf libisis3.so /scratch/isiscmake/isis/build/lib/libisis.so)
-endif()
```
Comment (Kelvinrr) [inline isis/cmake/cmake_install.cmake:49]: Not technically related to the issue, but part of this was removing hard-coded mentions to "ISIS3". This block doesn't seem to do much anymore (seems like old debug stuff?), so I simply removed it.

File: isis/CMakeLists.txt [hunk 2]
```diff
@@ -32,6 +32,11 @@ set(PACKAGE_NAME       "USGS ISIS")
 
 # Version number
 set(VERSION            "4.1.0")
+string(REPLACE "." ";" VERSION_LIST ${VERSION})
+list(GET VERSION_LIST 0 VERSION_MAJOR)
+list(GET VERSION_LIST 1 VERSION_MINOR)
+list(GET VERSION_LIST 2 ERSION_PATCH)
```
Comment (jessemapel) [inline isis/CMakeLists.txt]: missing a V on VERSION_PATCH

File: isis/CMakeLists.txt [hunk 3]
```diff
@@ -421,9 +428,9 @@ endif()
 
 # We will set up some links with these files at the end of the install process so
 #  make sure they are cleared at the start of the install process.
-install(CODE "EXECUTE_PROCESS(COMMAND rm -f ${CMAKE_INSTALL_PREFIX}/lib/libisis3.6.0${SO})")
-install(CODE "EXECUTE_PROCESS(COMMAND rm -f ${CMAKE_INSTALL_PREFIX}/lib/libisis3.6${SO})")
-install(CODE "EXECUTE_PROCESS(COMMAND rm -f ${CMAKE_INSTALL_PREFIX}/lib/libisis3.${SO})")
+install(CODE "EXECUTE_PROCESS(COMMAND rm -f ${CMAKE_INSTALL_PREFIX}/lib/libisis${VERSION}${SO})")
+install(CODE "EXECUTE_PROCESS(COMMAND rm -f ${CMAKE_INSTALL_PREFIX}/lib/libisis${VERSION_MAJOR}.${VERSION_MINOR}${SO})")
+install(CODE "EXECUTE_PROCESS(COMMAND rm -f ${CMAKE_INSTALL_PREFIX}/lib/libisis${VERSION_MAJOR}.${SO})")
```
Comment (jessemapel) [inline isis/CMakeLists.txt]: Having a "." before ${SO} is inconsistent here. It's that way in the original and it doesn't seem to have an impact? I would just remove the "." so it matches the install part.

**Comment by jessemapel** [comment]
Looks good to go with minor clean-up and a merge conflict fix.

File: isis/CMakeLists.txt [hunk 4]
```diff
@@ -68,7 +73,7 @@ option(isisTestData    "Directory containing IsisTestData"              OFF )
 option(testOutputDir   "Directory to store app test output folders"     OFF )
 option(buildCore       "Build the core ISIS modules"                    ON  )
 option(buildMissions   "Build the mission specific modules"             ON  )
-option(buildStaticCore "Build libisis3 static as well as dynamic"       OFF )
+option(buildStaticCore "Build libisis static as well as dynamic"       OFF )
```
Comment (krlberry) [inline isis/CMakeLists.txt]: Nitpickiest thing ever, but could you put a space back in at the end of the string so that the OFFs and ONs line up again?

**Review by jessemapel — APPROVED** [review]

File: isis/CMakeLists.txt (hunk 2 above)
Comment (Kelvinrr) [reply isis/CMakeLists.txt]: Wasn't being used so I simply removed it.

**Pull request merged**